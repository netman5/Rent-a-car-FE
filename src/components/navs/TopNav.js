import React from 'react';
import { NavLink } from 'react-router-dom';
import Button from '@mui/material/Button';
import { useDispatch } from 'react-redux';
import styles from './TopNav.module.css';

const TopNav = () => {
  const isLoggedIn = localStorage.getItem('isLoggedIn');
  const data = JSON.parse(localStorage.getItem('user'));
  const dispatch = useDispatch();

  const { user } = data || {};

  const handleLogout = () => {
    localStorage.removeItem('user');
    localStorage.removeItem('token');
    localStorage.removeItem('isLoggedIn');
    dispatch({ type: 'LOGGED_OUT' });
    window.history.pushState({}, '', '/');
    window.location.reload();
  };

  return (
    <div className={styles.topnav}>
      <h2>RentACar</h2>
      <div className={styles.auth}>
        {
        isLoggedIn ? (
          <div className={styles.userInfo}>
            <span>
              Hi
              {' '}
              {user.username}
            </span>

            <span>
              {' '}
              {user.email}
            </span>
            <Button type="button" onClick={handleLogout}>Logout</Button>
          </div>
        ) : (
          <>
            <Button variant="contained"><NavLink to="/sign_in" className={styles.login}>Login</NavLink></Button>
            <Button type="button"><NavLink to="/sign_up" className={styles.signup}>Register</NavLink></Button>
          </>
        )
      }

      </div>
    </div>
  );
};

export default TopNav;
