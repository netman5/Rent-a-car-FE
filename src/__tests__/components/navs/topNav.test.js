import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import TopNav from '../../../components/navs/TopNav';
import store from '../../../redux/configureStore';

it('Cars renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <TopNav />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
