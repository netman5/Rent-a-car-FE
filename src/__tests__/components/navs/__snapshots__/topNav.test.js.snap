// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cars renders correctly 1`] = `
<div
  className="topnav"
>
  <h2>
    RentACar
  </h2>
  <div
    className="auth"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <a
        className=""
        href="/sign_in"
        onClick={[Function]}
      >
        Login
      </a>
    </button>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <a
        className=""
        href="/sign_up"
        onClick={[Function]}
      >
        Register
      </a>
    </button>
  </div>
</div>
`;
